yoyodyne/train.py

```diff
@@ -36,59 +36,56 @@ def _get_logger(experiment: str, model_dir: str, log_wandb: bool) -> List:
 
 
 def _get_callbacks(
+    num_checkpoints: int = defaults.NUM_CHECKPOINTS,
+    checkpoint_metric: str = defaults.CHECKPOINT_METRIC,
     patience: Optional[int] = None,
-    save_top_k: int = defaults.SAVE_TOP_K,
-    val_metric: str = defaults.VAL_MODE,
-) -> List:
+    patience_metric: str = defaults.PATIENCE_METRIC,
+) -> List[callbacks.Callback]:
     """Creates the callbacks.
 
     We will reach into the callback metrics list to picks ckp_callback to find
     the best checkpoint path.
 
     Args:
-        patience (int, optional).
+        num_checkpoints (int, optional): number of checkpoints to save. To
+            save one checkpoint per epoch, use `-1`.
+        checkpoint_metric (string, optional): the
+        patience (int, optional): number of epochs with no
```

Comment:
1) I think we need a comment for `patience_metric`
2) Should patience be an attribute of the patience metric object? I think probably no given the design, but just want to double check this is intentional.

Reply:
Actually ignore 2)---this is synonymous to how we have a separate `num_checkpoints` variable. The `Metric` is only concerned about the metric specification, not the specific interval.

Reply:
+1 yeah